environment/action.yaml

```diff
@@ -0,0 +1,62 @@
+name: Environment Details
+description: Calculate Environment Details from Branch Identifier
+
+inputs:
+  environment:
+    description: Branch Identifier (either GitHub Branch or Environment Name)
+    required: true
+outputs:
+  github-branch: 
+    description: The GitHub Branch Name
+    value: ${{ steps.branch.outputs.github_branch }}
+  environment-name:
+    description: The AWS Environment Name
+    value: ${{ steps.branch.outputs.environment_name }}
+  environment-type:
+    description: The type of AWS Environment Identified
+    value: ${{ steps.branch.outputs.environment_type }}
+
+runs:
+  using: composite
+  steps:
+
+    - name: Get Branch Name
+      shell: bash
+      id: branch
+      run: |
+        # Get Environment and Branch information from provided input
+        shopt -s extglob
+        case ${{ inputs.environment }} in
+
+          ?(feature/)+([cb|CB|vt|VT])*)
+            environment_type=feature
+            environment_name=$(echo ${{ inputs.environment }} | sed 's/feature\///' | tr '[:upper:]' '[:lower:]')
+            github_branch=feature/$(echo ${{ inputs.environment }} | sed 's/feature\///' | tr '[:lower:]' '[:upper:]')
+            ;;
+
+          ?(release/)v+([0-9.]))
+            environment_type=release
+            environment_name=$(echo ${{ inputs.environment }} | sed 's/release\///' | tr '[:upper:]' '[:lower:]')
```

Comment:
Is 'environment_name' meant for use in Terraform for example? release goes to INT or PreProd

Reply:
No, it will return 'release' for a release. Deployment to higher environments will need to be pointed at those environments by the workflow that triggers the release.